### ISSUE TITLE
Mentions change to deleted

### ISSUE THREAD

**Issue by dannyuk1982** [flarum/core#2989]
## Bug Report

**Current Behavior**
When I mention a user in a numbered list, it always changes it to [deleted] the first time

**Steps to Reproduce**
1. Create a numbered list, mention a user, save

**Expected Behavior**
The user not to change to [deleted]

**Screenshots**
https://user-images.githubusercontent.com/7669885/127372618-643a90ac-acc3-4c2a-b27c-3586dc688873.mov

**Environment**
- Flarum version: 1.0.4
- Website URL: https://talk.lbbl.co.uk
- Webserver: apache
- Hosting environment: vps
- PHP version: 7.3.29
- Browser: Chrome 91.0.4472.164 and Safari 14.1.1

```
Output of "php flarum info", run this in terminal in your Flarum directory.

Flarum core 1.0.4
PHP version: 7.3.29
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, bz2, calendar, ctype, hash, filter, ftp, gettext, gmp, SPL, iconv, pcntl, readline, Reflection, session, standard, SimpleXML, sockets, mbstring, tokenizer, xml, mysqlnd, bcmath, curl, dba, dom, enchant, fileinfo, gd, imagick, imap, intl, json, ldap, exif, mysqli, odbc, PDO, pdo_mysql, PDO_ODBC, pdo_pgsql, pdo_sqlite, pgsql, Phar, posix, pspell, redis, soap, sodium, sqlite3, sysvmsg, sysvsem, sysvshm, tidy, xmlreader, xmlrpc, xmlwriter, xsl, zip, ionCube Loader, Zend OPcache
+------------------------------+---------+--------+
| Flarum Extensions            |         |        |
+------------------------------+---------+--------+
| ID                           | Version | Commit |
+------------------------------+---------+--------+
| flarum-suspend               | v1.0.0  |        |
| flarum-tags                  | v1.0.3  |        |
| fof-impersonate              | 1.0.0   |        |
| flarum-subscriptions         | v1.0.0  |        |
| fof-user-bio                 | 1.0.0   |        |
| justoverclock-welcomebox     | 1.2.8   |        |
| fof-default-user-preferences | 1.0.0   |        |
| askvortsov-pwa               | v3.0.3  |        |
| fof-oauth                    | 1.0.1   |        |
| flarum-bbcode                | v1.0.0  |        |
| flarum-emoji                 | v1.0.0  |        |
| flarum-lang-english          | v1.0.0  |        |
| flarum-likes                 | v1.0.0  |        |
| flarum-lock                  | v1.0.0  |        |
| flarum-markdown              | v1.0.1  |        |
| flarum-statistics            | v1.0.0  |        |
| fof-upload                   | 1.0.1   |        |
| fof-user-directory           | 1.0.2   |        |
| fof-split                    | 1.0.0   |        |
| fof-merge-discussions        | 1.0.0   |        |
| fof-frontpage                | 1.0.0   |        |
| fof-links                    | 1.0.0   |        |
| fof-username-request         | 1.0.0   |        |
| flarum-pusher                | v1.0.0  |        |
| fof-polls                    | 1.0.3   |        |
| fof-follow-tags              | 1.0.1   |        |
| flarum-mentions              | v1.0.0  |        |
| flarum-sticky                | v1.0.0  |        |
| fof-moderator-notes          | 1.0.0   |        |
+------------------------------+---------+--------+
Base URL: https://talk.lbbl.co.uk
Debug mode: ON
```

**Comment by dannyuk1982** [comment]
Any news on this? I think it's probably a Javascript error. I've also had it happen outside of numeric lists, can't just be me surely?

**Comment by SychO9** [comment]
The autocomplete dropdown should have loaded the user into the store, so it's odd that it can't find the user since the user object should be available. Needs to be investigated.

**Comment by dannyuk1982** [comment]
I've done a bit more testing, if I tag someone already on the list again then it works first time, otherwise it exhibits this behaviour. 

The tag, before saving, appears as:

`<span class="UserMention UserMention-deleted">@TimH</span>`

So seems like it defaults to being marked as deleted. If the user data is stored in a JS store somewhere it seems like it's not getting loaded in in time, which explains why it works the second time or with a user already tagged?

(Caveat: I know nothing about what JS Flarum uses or what architecture etc. it uses, so this coudl be nonsense!)

**Comment by matteocontrini** [comment]
I'm seeing this issue even without numbered lists and I'm able to reproduce it consistently, I would say every single time I use mentions.

**Comment by dannyuk1982** [comment]
I guess this isn't available yet? I just updated via composer but info says 1.0.4 still and it's still going it.

**Comment by SychO9** [comment]
it will be available in 1.1

**Comment by askvortsov1** [comment]
Ah for some reason I thought that got merged.
